tyssue/core/sheet.py

```diff
@@ -273,6 +275,75 @@ def extract_bounding_box(
         subsheet.reset_topo()
         return subsheet
 
+    def get_force_inference(self, coords=None):
+        """Measure force based on Brodland method.
+
+        g_gamma_matrix*tension_vector = 0
+        Equation is homogenous and to avoid tension_vectors = 0,
+        Construction and solve the constrained least-squares equation system
+        [[g_gamma_matrix.T g_gamma_matrix, C^T_1],
+         [C_1, 0]]
+         où C1 = {1....1}
+
+        """
+        if coords is None:
+            coords = self.coords
+        ndim = self.dim
+        dcoords = ["d" + c for c in coords]
+
+        self.get_extra_indices()
+        edges_index = self.east_edges
+        edges_index_opposite = self.west_edges
+
+        n_edges = len(edges_index)
+
+        srce, trgt = self.edge_df.loc[
+            edges_index, ['srce', 'trgt']].iteritems()
+        srce = srce[1].to_numpy()
+        trgt = trgt[1].to_numpy()
+
+        # Fill gamma matrix to measure tension
+        pos = (self.edge_df.loc[edges_index, dcoords].to_numpy() / np.repeat(np.linalg.norm(
+            self.edge_df.loc[edges_index, dcoords].to_numpy(), axis=1), ndim).reshape((n_edges, ndim)))
+
+        pos = np.concatenate((pos, -pos)).flatten()
+        row = np.concatenate((
+            np.array([np.arange(i * ndim, i * ndim + ndim)
+                      for i in srce]).flatten(),
+            np.array([np.arange(i * ndim, i * ndim + ndim)
+                      for i in trgt]).flatten())
+        )
+
+        col = np.concatenate(
+            (np.repeat(np.arange(n_edges), ndim), np.repeat(np.arange(n_edges), ndim)))
```

Comment:
If I understand correctly you solve for both east and west edges, but it should be possible to only solve for west_edges (plus free_edges maybe) and set the east_edges tension (or its opposite) for the west_edes?

Reply:
I resolve for east edges only, and after set the east_edges tension for the west_edges.

Reply:
but then why is the matrix this size? Maybe I miss something here

Reply:
The matrix size is (Ne/2, Nv*ndim)

Reply:
ok, so there is indeed something I did not get :)

one thing that might make things easier (and drop the list comprehensions) is to arrange the data a bit differently. Now you do (if I understand correctly) 

[x_s0, y_s0, z_s0, x_s1, y_s1, z_s1, ... x_si, y_si, z_si, ...]. where si is the index of the source vertex for edge i. Now if you use [x_s0, x_s1 ... x_si ..., y_s0, y_s1, ...], you can write  your rows as 
```py
row = np.concatenate(
    [srce+i*Ne for i in range(dim)] + [trgt+i*Ne for i in range(dim)]
)
```

and the cols are "just" `np.tile(np.arange(n_edges), dim*2))` or something approaching

Reply:
I don't have the same result with this method. Matrix is not fill with the same values in both method. 
Also, I think it's important to keep x,y,z coordinates together to solve the matrix after.